lib/Dialect/MSFT/MSFTOps.cpp

```diff
@@ -628,6 +628,32 @@ void MSFTModuleOp::print(OpAsmPrinter &p) {
   printModuleLikeOp<MSFTModuleOp>(*this, p, getParametersAttr());
 }
 
+LogicalResult MSFTModuleOp::verify() {
+  auto &body = getBody();
+  if (body.empty())
+    return success();
+
+  // Verify the number of block arguments.
+  auto type = getFunctionType();
+  auto numInputs = type.getNumInputs();
+  auto *bodyBlock = &body.front();
+  if (bodyBlock->getNumArguments() != numInputs)
+    return emitOpError("entry block must have")
+           << numInputs << " arguments to match module signature";
+
+  // Verify the number of block arguments.
+  if (bodyBlock->getNumArguments() != numInputs)
```

Comment:
This code appears to be identical to the code on line 640. Copy & paste & paste error?

Reply:
Yeah, c&p error from the HW side